crates/amaru-ledger/src/rules/transaction/outputs.rs

```diff
@@ -50,12 +53,16 @@ pub enum InvalidOutput {
     UncategorizedError(String),
 }
 
-pub fn execute(
+pub fn execute<C>(
+    context: &mut C,
     protocol_parameters: &ProtocolParameters,
     network: &Network,
     outputs: Vec<MintedTransactionOutput<'_>>,
-    yield_output: &mut impl FnMut(u64, TransactionOutput),
-) -> Result<(), InvalidOutputs> {
+    yield_output: &mut impl FnMut(&mut C, u64, TransactionOutput),
```

Comment:
Doesn't sound right. Why are we yielding back the context to a caller that already has it?

Reply:
Because of ownership issues -- the `outputs::execute` function needs to use the `context` to allow supplemental datums. But then the yield output function can't also call the context, so we pass it to the argument function.

I'd argue that's a code smell, but I just wanted to get it "working" for now

Reply:
I am not convinced. "Getting it working for now" isn't a valid excuse 😶 (I am exaggerating of course, no big deal here but still, not convinced!). There's no such thing as a "temporary work-around". So I am afraid this would stay. 

At the very least, please add a TODO.